### PR TITLE
Hopefully fix shortener issues

### PR DESCRIPTION
**Pull request by leogrim**
@andrewconner 
(if this works, we can get rid of some of the callback routes / methods)

### CODE REVIEW

**Comment by andrewconner** [comment]
Should be good — never seen this code before though :)

**Comment by andrewconner** [comment]
Let's try it.

**Comment by leogrim** [comment]
Actually it wasn't, additional commit coming.

File: kifi-backend/modules/shoebox/app/com/keepit/integrity/UriIntegrityPlugin.scala [hunk 1]
```diff
@@ -384,3 +383,26 @@ class UriIntegrityPluginImpl @Inject() (
...
+        case Some(url) => keepWithTitle.copy(url = url.url, urlId = url.id.get)
+      }
+    } else keepWithTitle
+  }
+
+  def improveKeepsSafely(uri: NormalizedURI)(implicit session: RWSession): Unit = {
+    keepRepo.getByUri(uri.id.get).foreach { keep =>
+      val betterKeep = improveKeepSafely(uri, keep)
+      if (betterKeep != keep) { keepRepo.save(betterKeep) }
+    }
```
Comment (leogrim) [inline kifi-backend/modules/shoebox/app/com/keepit/integrity/UriIntegrityPlugin.scala:402]: @andrewconner

**Pull request merged**